llm_claude/__init__.py

```diff
@@ -1,65 +1,103 @@
+            raise ValueError("temperature must be in range 0.0-1.0")
+        return temperature
+
+    @field_validator("top_p")
+    def validate_top_p(cls, top_p):
+        if top_p is not None and not (0.0 <= top_p <= 1.0):
+            raise ValueError("top_p must be in range 0.0-1.0")
+        return top_p
+
+    @field_validator("top_k")
+    def validate_top_k(cls, top_k):
+        if top_k is not None and top_k <= 0:
+            raise ValueError("top_k must be a positive integer")
+        return top_k
 
 
 class Claude(llm.Model):
     needs_key = "claude"
     key_env_var = "ANTHROPIC_API_KEY"
     can_stream = True
 
-    class Options(llm.Options):
-        max_tokens_to_sample: Optional[int] = Field(
-            description="The maximum number of tokens to generate before stopping",
-            default=10_000,
-        )
-
-        @field_validator("max_tokens_to_sample")
-        def validate_length(cls, max_tokens_to_sample):
-            if not (0 < max_tokens_to_sample <= 1_000_000):
-                raise ValueError("max_tokens_to_sample must be in range 1-1,000,000")
-            return max_tokens_to_sample
+    class Options(_ClaudeOptions):
+        ...
 
     def __init__(self, model_id):
         self.model_id = model_id
 
     def generate_prompt_messages(self, prompt, conversation):
+        messages = []
         if conversation:
             for response in conversation.responses:
-                yield self.build_prompt(response.prompt.prompt, response.text())
-
-        yield self.build_prompt(prompt)
-
-    def build_prompt(self, human, ai=""):
-        return f"{HUMAN_PROMPT} {human}{AI_PROMPT}{ai}"
+                messages.append({"role": "user", "content": response.prompt.prompt})
+                messages.append({"role": "assistant", "content": response.text()})
+        messages.append({"role": "user", "content": prompt})
+        return messages
 
     def execute(self, prompt, stream, response, conversation):
         anthropic = Anthropic(api_key=self.get_key())
-
-        prompt_str = "".join(self.generate_prompt_messages(prompt.prompt, conversation))
-
-        completion = anthropic.completions.create(
+        messages = self.generate_prompt_messages(prompt.prompt, conversation)
+        completion = anthropic.messages.create(
             model=self.model_id,
-            max_tokens_to_sample=prompt.options.max_tokens_to_sample,
-            prompt=prompt_str,
+            max_tokens=prompt.options.max_tokens,
+            messages=messages,
             stream=stream,
         )
         if stream:
             for comp in completion:
-                yield comp.completion
+                if hasattr(comp, "content_block"):
+                    response = comp.content_block.text
+                    yield response
+                elif hasattr(comp, "delta"):
+                    if hasattr(comp.delta, "text"):
+                        yield comp.delta.text
```

Comment:
this block of code is pretty fast and loose, based on their documented SSE format [here](https://docs.anthropic.com/claude/reference/messages-streaming). I mostly just inspected the python objects coming out of the completion stream and used this to get the raw text deltas. there's probably a more principled way to do it, happy to rewrite if that's a concern